tests/test_batch.py

```diff
@@ -183,48 +236,52 @@ def custom_get_batch_mock(request, _):
 
         # Reset history so that we can count calls later
         mock_request.reset_mock()
+        mock_request.register_uri(
+            "POST",
+            f"/core-fast/api/v1/batches/{batch.id}/jobs",
+            json=load_mock_batch_json_response,
+        )
         batch.add_jobs(
             [{"runs": self.n_job_runs}, {"runs": self.n_job_runs}],
             wait=True,
         )
 
-        assert len(batch.ordered_jobs) == 2
-        assert (
-            mock_request.call_count == 5
-        )  # 1 call to add jobs and 4 get batch calls until jobs are DONE
+        assert mock_request.call_count == 12
+
         assert (
             mock_request.request_history[0].url
-            == f"{self.sdk._client.endpoints.core}/api/v1/batches/{batch.id}/jobs"
+            == f"{self.sdk._client.endpoints.core}/api/v1/batches/00000000-0000-0000-0000-000000000002/jobs"
```

Comment:
There was some unsavory mocked behavior I could not pinpoint, but compared IDs were not equal. Really struggled to pinpoint why